rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/release/Sw360ReleaseService.java

```diff
@@ -1419,4 +1426,64 @@ private void addLicenseInfoResultToJsonSerializableLicensesList(LicenseInfoParsi
                 throw new IllegalArgumentException("Unknown LicenseInfoRequestStatus: " + licenseInfoResult.getStatus());
         }
     }
+
+    public Map<String, Object> getReleaseLicenseFileListInfo(Release rel, User sw360User) throws TException{
+        ThriftClients thriftClients = new ThriftClients();
+        LicenseInfoService.Iface licenseClient = thriftClients.makeLicenseInfoClient();
+        final Predicate<Attachment> isCLI = attachment -> AttachmentType.COMPONENT_LICENSE_INFO_XML.equals(attachment.getAttachmentType())
+                || AttachmentType.COMPONENT_LICENSE_INFO_COMBINED.equals(attachment.getAttachmentType());
+        Set<LicenseNameWithText> licenseNameWithTexts = new HashSet<LicenseNameWithText>();
+        Map<String, Object> successResponse = new HashMap<>();
+        List<Attachment> filteredAttachments = CommonUtils.nullToEmptySet(rel.getAttachments()).stream().filter(isCLI).collect(Collectors.toList());
+        if (filteredAttachments.size() > 1) {
+            Predicate<Attachment> isApprovedCLI = attachment -> CheckStatus.ACCEPTED.equals(attachment.getCheckStatus());
+            filteredAttachments = filteredAttachments.stream().filter(isApprovedCLI).collect(Collectors.toList());
+        }
+        if (filteredAttachments.size() == 1 && filteredAttachments.get(0).getFilename().endsWith(SW360Constants.XML_FILE_EXTENSION)) {
+            final Attachment filteredAttachment = filteredAttachments.get(0);
+            final String attachmentContentId = filteredAttachment.getAttachmentContentId();
+            try {
+                List<LicenseInfoParsingResult> licenseResults = licenseClient.getLicenseInfoForAttachment(rel, attachmentContentId, false, sw360User);
+                if (CommonUtils.isNotEmpty(licenseResults) && LicenseInfoRequestStatus.SUCCESS.equals(licenseResults.get(0).getStatus())) {
+                    licenseNameWithTexts = licenseResults.get(0).getLicenseInfo().getLicenseNamesWithTexts();
+                    if (CommonUtils.isNotEmpty(licenseNameWithTexts)) {
+                        List<Map<String, Object>> licenseToSourceData = new ArrayList<>();
+                        for (LicenseNameWithText license : licenseNameWithTexts) {
+                            Map<String, Object> licenseDetails = new HashMap<>();
+                            String fileList = String.join(",", license.getSourceFiles());
+                            List<String> resultList = Arrays.asList(fileList.split("\\n"));
+                            licenseDetails.put("licName", nullToEmptyString(license.getLicenseName()));
+                            licenseDetails.put("licSpdxId", nullToEmptyString(license.getLicenseSpdxId()));
+                            licenseDetails.put("srcFiles", nullToEmptyString(resultList));
```

Comment:
Upon test, `nullToEmptyString` is converting the `List<String>` back to `String` which is not desired. Please use a different check.